lib/datadog/statsd.rb

```diff
@@ -34,8 +34,8 @@ class Connection
       attr_reader :socket_path
 
       def initialize(host, port, socket_path, logger)
-        @host = host || DEFAULT_HOST
-        @port = port || DEFAULT_PORT
+        @host = host || ENV.fetch('DD_AGENT_HOST', nil) || DEFAULT_HOST
```

Comment:
Shouldn't this be ENV vars > parameters > defaults ? Having to put `nil`s there seems odd to me

Reply:
actually we discussed the priority order (params, env vars, defaults) with the team and there were some different points of view regarding this, finally we agreed to prioritize parameters over env vars and this logic has been/is being implemented in all other languages dogstatsd clients.
The main point of prioritizing params over env vars it that it would be odd to override the user arguments especially when he/she doesn't look into the library code.
We can still discuss that if you find it unconvincing, and I suggest we update the README file anyway to mention the order of priority we agree on :)

Reply:
What is use case behind this change in the first place ? Is it to make it easier to configure in containerized environments ?

Reply:
yes, you can have a look at the go client PR here https://github.com/DataDog/datadog-go/pull/78
and the very first purpose is origin detection using the entity_id tag

Reply:
My reasoning was that both the environment variable and the configuration at the code level are defined by the user. If both are set ENV vars usually trumps static configuration (like with our agent).
But if the conclusion from your team differs, that's perfectly fine. Both are working solutions, it's mostly a opinion 😄

Reply:
Thank you @arbll for pointing out 😄 
if you approve the PR I guess we can merge, cc @masci